packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -129,175 +42,260 @@ if (__DEV__) {
+        didWarnAboutUnsafeLifecycles.add(fiber.type);
+      });
+
+      pendingUNSAFE_ComponentWillReceivePropsWarnings = [];
+    }
+
+    const componentWillUpdateUniqueNames = new Set();
+    if (pendingComponentWillUpdateWarnings.length > 0) {
+      pendingComponentWillUpdateWarnings.forEach(fiber => {
+        componentWillUpdateUniqueNames.add(
+          getComponentName(fiber.type) || 'Component',
+        );
+        didWarnAboutUnsafeLifecycles.add(fiber.type);
+      });
+
+      pendingComponentWillUpdateWarnings = [];
+    }
+
+    const UNSAFE_componentWillUpdateUniqueNames = new Set();
+    if (pendingUNSAFE_ComponentWillUpdateWarnings.length > 0) {
+      pendingUNSAFE_ComponentWillUpdateWarnings.forEach(fiber => {
+        UNSAFE_componentWillUpdateUniqueNames.add(
+          getComponentName(fiber.type) || 'Component',
+        );
+        didWarnAboutUnsafeLifecycles.add(fiber.type);
+      });
+
+      pendingUNSAFE_ComponentWillUpdateWarnings = [];
+    }
+
+    // Finally, we flush all the warnings
+    // UNSAFE_ ones before the deprecated ones, since they'll be 'louder'
+    if (UNSAFE_componentWillMountUniqueNames.size > 0) {
+      const sortedNames = setToSortedString(
+        UNSAFE_componentWillMountUniqueNames,
+      );
       warningWithoutStack(
         false,
-        'Expected to find a StrictMode component in a strict mode tree. ' +
-          'This error is likely caused by a bug in React. Please file an issue.',
+        'Using UNSAFE_componentWillMount in strict mode is not recommended and may indicate bugs in your code. ' +
+          'See https://fb.me/react-async-component-lifecycle-hooks for details.\n\n' +
+          '* Move code with side effects to componentDidMount, and set initial state in the constructor.\n' +
+          '\nPlease update the following components: %s',
+        sortedNames,
       );
-      return;
     }
 
-    // Dedup strategy: Warn once per component.
-    // This is difficult to track any other way since component names
-    // are often vague and are likely to collide between 3rd party libraries.
-    // An expand property is probably okay to use here since it's DEV-only,
-    // and will only be set in the event of serious warnings.
-    if (didWarnAboutUnsafeLifecycles.has(fiber.type)) {
-      return;
+    if (UNSAFE_componentWillReceivePropsUniqueNames.size > 0) {
+      const sortedNames = setToSortedString(
+        UNSAFE_componentWillReceivePropsUniqueNames,
+      );
+      warningWithoutStack(
+        false,
+        'Using UNSAFE_componentWillReceiveProps in strict mode is not recommended ' +
+          'and may indicate bugs in your code. ' +
+          'See https://fb.me/react-async-component-lifecycle-hooks for details.\n\n' +
+          '* Move data fetching code or side effects to componentDidUpdate.\n' +
+          "* If you're updating state whenever props change, " +
+          'refactor your code to use memoization techniques, or move it to ' +
```

Comment:
grammar nit: This "," is unnecessary and breaks the flow of the sentence.

(I know, I know...)

Reply:
dammit, missed this one.

Reply:
ok fixed, updated screenshot